frontend/src/components/application/attachments/DeleteAttachmentButton.tsx

```diff
@@ -0,0 +1,44 @@
+"use client";
+
+import { Attachment } from "src/types/attachmentTypes";
+
+import { RefObject } from "react";
+import { ModalRef, ModalToggleButton } from "@trussworks/react-uswds";
+
+import { USWDSIcon } from "src/components/USWDSIcon";
+
+interface AttachmentDeleteButtonProps {
+  buttonText: string;
+  file: Attachment;
+  handleDeleteAttachment: (
+    application_attachment_id: string,
+    attachmentToDeleteName: string,
+  ) => void;
+  modalRef: RefObject<ModalRef | null>;
+}
+
+export const DeleteAttachmentButton = ({
+  buttonText,
+  file,
+  handleDeleteAttachment,
+  modalRef,
+}: AttachmentDeleteButtonProps) => {
+  return (
+    <ModalToggleButton
+      disabled={false}
+      modalRef={modalRef}
+      opener
+      className="usa-nav__link font-sans-2xs display-flex text-normal border-0"
+      data-testid="delete-attachment-modal-toggle-button"
+      onClick={() =>
+        handleDeleteAttachment(file.application_attachment_id, file.file_name)
```

Comment:
this is a little misleading - seems like the purpose of this function is to mark a file for deletion within the list rather than to actually delete the attachment. If I'm understanding right, can we rename?